### DISCUSSION TITLE
Auto export to svg/integration with obsidian.

### DISCUSSION THREAD

**Discussion by FAUSTINOCRUZELIS** [flxzt/rnote#1133] [Ideas]
I think Rnote is (for handwriting math and extensive notes) better than Excallydraw, so the existence of some way for easy integration to Obsidian could improve the potential of handwriting for project development.
Auto export like autosave in MS Word but to a diferent file (svg, png, jpg,... other) as a shortcut(for manual export preconfigured)/notification in the top bar (with some personalized settings).
Other option could be a module for obsidian to read/preview .rnote files inserted in MarkDown obsidian's notes, and redirect to open/edit it on rnote.
![imagen](https://github.com/flxzt/rnote/assets/76847804/80ca05e0-c37b-47ab-ba03-266ce5b75dc2)

**Comment by flxzt** [comment]
If you select something, and do a copy (either through Ctrl+C or through the menu) it automatically copies as Svg, Png and Rnote's custom Copy+Paste format.

Unfortunately Obsidian currently does not support pasting Svg's (I even opened a feature request for it in the forums, but there was no response yet..) but it should work to paste into the document as PNG.

**Comment by schmoelder** [comment]
> Other option could be a module for obsidian to read/preview .rnote files inserted in MarkDown obsidian's notes, and redirect to open/edit it on rnote.

That would be great. This way, I could simply attach an rnote to the actuall note and edit it when required.

**Reply by kalinkochnev** [reply]
I am very interested in implementing this functionality. I am looking into how difficult it would be to build a proof of concept.

**Reply by gregorianrants** [reply]
here is the script i wrote to do sync with a pdf, it was pretty simple, watches the file then every time it changes runs the rnote-cli export command.

just change suffix of the output file to have it mirrored in svg

if the version of rnote you have has the rnote-cli, if you run this script in the folder your rnote file is in then you will have pdf that mirrors it and updates on save.

not sure if this is exactly what you are looking for but having rnote files in a format that can be put in md is i think part of the puzzle.

```
import time
import pathlib
import subprocess
import os

print(os.getcwd())

from watchdog.events import FileSystemEvent, FileSystemEventHandler,PatternMatchingEventHandler
from watchdog.observers import Observer


class MyEventHandler(PatternMatchingEventHandler):
    def __init__(self,patterns):
        super().__init__(patterns=patterns)


    def on_modified(self, event: FileSystemEvent) -> None:
        print(event)
        
        filename = pathlib.Path(event.src_path).stem
        print(filename)
        output_file = f'{filename}.pdf'
        input_file = f'{filename}.rnote'

        #subprocess.run([f'rnote-cli export doc --on-conflict overwrite -o {filename}.pdf {filename}.rnote'],cwd=os.path.realpath(__file__))
        subprocess.run(['rnote-cli','export','doc','--on-conflict','overwrite','-o',output_file,input_file])
        
        #result = subprocess.run(["dir"], shell=True, capture_output=True, text=True) 
        #print(result.stdout) 


event_handler = MyEventHandler(patterns=['*.rnote'])
observer = Observer()
observer.schedule(event_handler, ".", recursive=False)
observer.start()
try:
    while True:
        time.sleep(1)
finally:
    observer.stop()
    observer.join()
```
would probably be even easier to do this in nodejs which obsidian uses for plugins, i dont know much about the packaging of plugins so have just stuck with this script.

**Reply by kalinkochnev** [reply]
~~@gregorianrants I think alternatively it may be possible to compile rnote-cli into wasm, create an obsidian extension, and and automatically render `*.rnote` files without creating new files.~~

After a first attempt WASM doesn't seem immediately doable. `rnote-engine` has a few usages of the `gio` package and both `rnote-engine` and `rnote-compose` use `piet_cairo`. If [`piet_common`](https://github.com/linebender/piet/tree/main) is used then maybe `piet_web` vs `piet_cairo` can be selected at compile time.    

On an unrelated note I think it would be nice if we could render only certain regions of an rnote file.

**Reply by gregorianrants** [reply]
I personally dont know anything about assembling into wasm or writing obsidian plugins. if the features you are talking about became available in a plugin though i would love to use them.

the method i used is similar to how the xournal plugin works. you may find it interesting to have a play with. it uses the xournal cli which is pretty similar to the rnote one in what it can do re exporting.

**Reply by kalinkochnev** [reply]
Thanks @gregorianrants! I'll look into it

**Comment by der-wasguckstdu-dreier** [comment]
> I think Rnote is (for handwriting math and extensive notes) better than Excallydraw, so the existence of some way for easy integration to Obsidian could improve the potential of handwriting for project development. Auto export like autosave in MS Word but to a diferent file (svg, png, jpg,... other) as a shortcut(for manual export preconfigured)/notification in the top bar (with some personalized settings). Other option could be a module for obsidian to read/preview .rnote files inserted in MarkDown obsidian's notes, and redirect to open/edit it on rnote. ![imagen](https://private-user-images.githubusercontent.com/76847804/345756600-80ca05e0-c37b-47ab-ba03-266ce5b75dc2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzQ3OTM3NDYsIm5iZiI6MTc3NDc5MzQ0NiwicGF0aCI6Ii83Njg0NzgwNC8zNDU3NTY2MDAtODBjYTA1ZTAtYzM3Yi00N2FiLWJhMDMtMjY2Y2U1Yjc1ZGMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzI5VDE0MTA0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVhZjZiZjc4ZjNmYzUyMzI5MjJiNzUzMTA5ZjJjOGUyZDA3YTEwMWFiMDQ5OWRjMTAwMDYxMTc4NWZkMTNmNmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.EHylHoXFoZQFKEKC9xWka0nOZ4TUcB3AjzPXKKKjf88)

Hello, I found rNote while searching for an excalidraw alternative. While excalidraw is a great plugin, it just starts lagging before the lecture ends. Its not suitable for handwriting. Did you find a good way to automatically integrate rNote into obsidian?

**Comment by gregorianrants** [comment]
I have written a python script that automatically exports to pdf on save, can easily be adapted to svg.  I chose pdf rather than svg as pdfs automatically refresh on obsidian when the document changes.  just wrote it last night but will put it on github and share link here once i have done so.

for me combining rnote with obsidian is what will finally allow me to move away from one note.
